core/modrulestore.go

```diff
@@ -121,11 +121,21 @@ func (s *ModRuleStore) Delete(namespace string, name string) {
 func (s *ModRuleStore) getMatchingModRuleStoreItems(namespace string, modRuleType v1beta1.ModRuleType, jsonv interface{}) []*ModRuleStoreItem {
 	modRules := []*ModRuleStoreItem{}
 
-	if namespaceModRules, ok := s.modRuleListMap[namespace]; ok {
-		for _, mrsi := range namespaceModRules {
-			if mrsi.modRule.Spec.Type == modRuleType && mrsi.IsMatch(jsonv) {
-				modRules = append(modRules, mrsi)
-			}
+	potentialRules := []*ModRuleStoreItem{}
+	for _, mrsi := range s.modRuleListMap[""] {
+		if namespace == "" && mrsi.compiledTargetNamespaceRegex == nil ||
```

Comment:
Please replace with `if (namespace == "" && mrsi.compiledTargetNamespaceRegex == nil) || (mrsi.compiledTargetNamespaceRegex != nil && mrsi.compiledTargetNamespaceRegex.Match([]byte(namespace)))`, using parenthesis to demonstrate explicit intent.

Reply:
👍